polars/polars-lazy/polars-plan/src/dsl/function_expr/list.rs

```diff
@@ -226,15 +226,7 @@ pub(super) fn take(args: &[Series], null_on_oob: bool) -> PolarsResult<Series> {
     let idx = &args[1];
     let ca = ca.list()?;
 
-    if idx.len() == 1 {
```

Comment:
You can keep the `fast_path` but set this condition

```
if idx.len() == 1 && null_on_oob
```

Reply:
Thanks @ritchie46 - should have been able to figure that one out, d'oh.